.dotnet.azure/sdk/openai/Azure.AI.OpenAI/src/Custom/VectorStores/AzureVectorStoreClient.Protocol.cs

```diff
@@ -20,27 +20,15 @@ public override IEnumerable<ClientResult> GetVectorStores(int? limit, string ord
         return PageCollectionHelpers.Create(enumerator);
     }
 
-    public override async Task<ClientResult> CreateVectorStoreAsync(BinaryContent content, RequestOptions options = null)
-    {
-        using PipelineMessage message = CreateCreateVectorStoreRequest(content, options);
-        return ClientResult.FromResponse(await Pipeline.ProcessMessageAsync(message, options).ConfigureAwait(false));
-    }
-
-    public override ClientResult CreateVectorStore(BinaryContent content, RequestOptions options = null)
-    {
-        using PipelineMessage message = CreateCreateVectorStoreRequest(content, options);
-        return ClientResult.FromResponse(Pipeline.ProcessMessage(message, options));
-    }
-
-    public override async Task<ClientResult> GetVectorStoreAsync(string vectorStoreId, RequestOptions options)
+    internal override async Task<ClientResult> GetVectorStoreAsync(string vectorStoreId, RequestOptions options)
     {
         Argument.AssertNotNullOrEmpty(vectorStoreId, nameof(vectorStoreId));
 
         using PipelineMessage message = CreateGetVectorStoreRequest(vectorStoreId, options);
         return ClientResult.FromResponse(await Pipeline.ProcessMessageAsync(message, options).ConfigureAwait(false));
     }
 
-    public override ClientResult GetVectorStore(string vectorStoreId, RequestOptions options)
+    internal override ClientResult GetVectorStore(string vectorStoreId, RequestOptions options)
```

Comment:
Per our offline conversation about continuation tokens, it feels to me like this is the natural place to address the use-case you're interested in @chschrae.  We could make this method public and have it return the relevant LRO subclient.  Would that provide the feature support you're looking for?

@KrzysztofCwalina, would you be OK with this approach?

Reply:
Per offline discussion, it looks like we have tentative agreement around adding overloads of the static `Rehydrate` methods to address the user scenario instead of the approach proposed in the comment above.